win/rl/pynethack.cc

```diff
@@ -117,28 +117,32 @@ class Nethack
                 py::object inv_oclasses, py::object inv_strs)
     {
         std::vector<ssize_t> dungeon{ ROWNO, COLNO - 1 };
-        obs_.glyphs = checked_conversion<int16_t>(std::move(glyphs), dungeon);
-        obs_.chars = checked_conversion<uint8_t>(std::move(chars), dungeon);
-        obs_.colors = checked_conversion<uint8_t>(std::move(colors), dungeon);
-        obs_.specials =
-            checked_conversion<uint8_t>(std::move(specials), dungeon);
-        obs_.blstats = checked_conversion<long>(std::move(blstats),
-                                                { NLE_BLSTATS_SIZE });
-        obs_.message =
-            checked_conversion<uint8_t>(std::move(message), { 256 });
+        obs_.glyphs = checked_conversion<int16_t>(glyphs, dungeon);
```

Comment:
Why do we need to remove std::move?

Reply:
Good question.

A moved-from object in C++ becomes "empty", hence the `py_buffers_` assignment further down wouldn't be sensible. By removing the `std::move` we copy the `py::object`s, which increases their refcounts. Strictly speaking that's not necessary here, but this isn't performance-relevant code